tests/test_ble_interface_core.py

```
@@ -4971,6 +4980,249 @@ def test_find_device_direct_connect_without_discovery_manager() -> None:
+
+
+def test_invoke_safe_execute_compat_skips_callable_only_after_positional_failure() -> None:
+    """Positional safe_execute failures should not trigger a second handler invocation."""
+
+    calls: list[tuple[tuple[object, ...], dict[str, object]]] = []
+    handler_runs: list[str] = []
+    fallbacks: list[str] = []
+
+    def _handler_thunk() -> None:
+        handler_runs.append("run")
+        raise RuntimeError("handler failed")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)